compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -819,11 +813,13 @@ public void visit(BLangMatchStatement matchStatement) {
             }
             this.resetErrorThrown();
             analyzeNode(matchClause, env);
-            allClausesReturns = allClausesReturns && this.matchClauseReturns;
-            containsLastPatternInStatement =
-                    containsLastPatternInStatement || (matchClause.matchGuard == null && this.hasLastPatternInClause);
+            allClausesReturns = allClausesReturns && this.statementReturns;
+            resetStatementReturns();
+            matchStatement.hasLasPattern = matchStatement.hasLasPattern
+                    || (matchClause.matchGuard == null && matchClause.hasLastPatternInClause);
```

Comment:
I'm not sure if checking `matchGuard == null` will always produce correct results. See https://github.com/ballerina-platform/ballerina-lang/issues/27603

Reply:
I will check matchGuard related issues separately.